File: packages/shared/src/modes.ts
import type { GameContentType, SinglePlayerGameMode } from "./types";

export type SinglePlayerModeDefinition = {
  id: SinglePlayerGameMode;
  label: string;
  puzzleLabel: string;
  contentType: GameContentType;
};

export const SINGLE_PLAYER_MODE_DEFINITIONS: Record<
  SinglePlayerGameMode,
  SinglePlayerModeDefinition
> = {
  daily: {
    id: "daily",
    label: "每日题",
    puzzleLabel: "今日每日题",
    contentType: "character",
  },
  random: {
    id: "random",
    label: "随机题",
    puzzleLabel: "随机题",
    contentType: "character",
  },
};

export const isSinglePlayerGameMode = (
  value: string,
): value is SinglePlayerGameMode => value in SINGLE_PLAYER_MODE_DEFINITIONS;

File: apps/api/internal/game/modes.go
package game

// SinglePlayerModeDefinition 对应 shared 的 SinglePlayerModeDefinition。
type SinglePlayerModeDefinition struct {
	ID           string `json:"id"`
	Label        string `json:"label"`
	PuzzleLabel  string `json:"puzzleLabel"`
	ContentType  string `json:"contentType"`
}

// SinglePlayerModeDefinitions 对应 shared 的 SINGLE_PLAYER_MODE_DEFINITIONS。
var SinglePlayerModeDefinitions = map[string]SinglePlayerModeDefinition{
	"daily": {
		ID:          "daily",
		Label:       "每日题",
		PuzzleLabel: "今日每日题",
		ContentType: "character",
	},
	"random": {
		ID:          "random",
		Label:       "随机题",
		PuzzleLabel: "随机题",
		ContentType: "character",
	},
}

// IsSinglePlayerGameMode 对应 shared 的 isSinglePlayerGameMode。
func IsSinglePlayerGameMode(value string) bool {
	_, ok := SinglePlayerModeDefinitions[value]
	return ok
}
